src/ControlSystem/Trigger.hpp

```diff
@@ -143,6 +143,55 @@ class Trigger : public DenseTrigger {
 
     const bool triggered = time == *next_trigger_;
     if (triggered) {
+      // We must do this next check to ensure the functions of time have been
+      // updated. This is expected to run when time == the old function of time
+      // expiration time, a.k.a. the update time. At this point, it isn't
+      // sufficient that all the measurement timescales have been updated. We
+      // must ensure the functions of time have been updated as well because the
+      // next_measurement() function below uses the minimum expiration time of
+      // the functions of time when calculating the next measurement. The reason
+      // we have to do this complicated checking is because the functions of
+      // time are valid at the expiration times, so we can't just use a
+      // domain::functions_of_time_are_ready() call. That would just return true
+      // regardless of whether the functions of time have been updated. So we
+      // must check that the minimum expiration time is after the current time.
+      // If it is equal to the current time, then at least one of the functions
+      // of time hasn't been updated yet and we must wait until they are all
+      // updated so we set a callback.
+      if (must_check_fot_validity_) {
```

Comment:
Why is this check done conditionally?  It looks like `next_measurement` always needs it to be true, and it doesn't look terribly expensive.